core/shared/src/main/scala/fs2/internal/CompileScope.scala

```diff
@@ -101,12 +127,34 @@ private[internal] final class CompileScope[F[_]] private (val id: Token, private
         // This scope is already closed so try to promote the open to an ancestor; this can fail
         // if the root scope has already been closed, in which case, we can safely throw
         self.parent match {
-          case Some(parent) => parent.open
+          case Some(parent) => parent.open(interruptible)
           case None => F.raiseError(throw new IllegalStateException("cannot re-open root scope"))
         }
     }
   }
 
+  def acquireResource[R](fr: F[R], release: R => F[Unit]): F[Either[Throwable, (R, Token)]] = {
+    val resource = Resource.create
+    F.flatMap(register(resource)) { mayAcquire =>
+      if (!mayAcquire) F.raiseError(Interrupted(id, 0)) // todo: not sure if we shall signal by interrupted anymore
```

Comment:
TODO?

Reply:
Oh I see -- we used `Interrupted` to signal this condition before but now `Interrupted` has meaning that's specific to `interruptWhen`. How about a new internal exception type here, `AcquireAfterScopeClosed` or something?

Reply:
kk makes sense